### ISSUE TITLE
Can't setup on Heroku with ClearDB MySQL Database

### ISSUE THREAD

**Issue by lnhrdt** [dz0ny/heroku-docker-piwik#3]
I can't seem to deploy to Heroku and setup the database properly.

I setup a new app on Heroku, leaving the default config variables. Then I added the addon "ClearDB MySQL Database".

![ClearDB MySQL Database addon](https://cloud.githubusercontent.com/assets/4040241/8174969/5c2180d8-1398-11e5-8542-a782ed7e0f84.png)

It populates its `CLEARDB_DATABASE_URL` with the connection information so I parsed it and copied the info into `DB_HOST`, `DB_NAME`, `DB_PASSWORD`, `DB_USERNAME`.

![copy config](https://cloud.githubusercontent.com/assets/4040241/8175070/b17f5860-1399-11e5-9f85-5ad171fdb69c.png)

However when I open the app (Piwik) I don't get to the normal welcome configuration/setup screen, I just get an error.

![Piwik error](https://cloud.githubusercontent.com/assets/4040241/8175095/0603c236-139a-11e5-882f-a394a20b3ccb.png)

I've skimmed the logs on Heroku but I haven't been able to figure out what's up.

**Comment by dz0ny** [comment]
Make sure that env is reflected in vendor/piwik/piwik/config/config.ini.php, this file is only generated during build and does not reflect runtime changes.

**Comment by ghost** [comment]
Having the same issue here. Is there a way to re-build the app after changing the DB settings?

**Comment by ewwink** [comment]
I have created pull request to fix db or you can deploy directly from my repo https://github.com/ewwink/heroku-docker-piwik

**Comment by TheNotary** [comment]
This is getting overly complicated quick, but I think I have an optimal solution.  

I'm not a ClearDB MySQL user, but after some research it seems the first thing you should do when you setup the addon, is make the env variable named DATABASE_URL (see [here](https://devcenter.heroku.com/articles/cleardb)).  Let's assume all users of Piwik just do that...  Btw, if you're using dokku, the DATABASE_URL will be setup for you automatically.

Given the above, the ideal solution seems to be to have Piwik look up the database info from the DATABASE_URL.  

Edit:
The furthest I've gotten with this is in #10 which seems to just be a problem of getting the config file into a format that piwik wants it in.  Maybe this env var interpolation needs to take place in `entrypoint.sh`??

Edit:
After doing much more digging, it seems a selling point of Piwik PRO is to disable automation of the install in the open source version, and make sure those users install Piwik over the web UI which generates a config.ini.php file and additionally populate the mysql database.  https://github.com/piwik/piwik-package/issues/40

**Comment by TheNotary** [comment]
After much effort and tweaking, I was unable to make Heroku behave the same way dokku does.  For this reason I started [piwik-dokku](https://github.com/thenotary/piwik-dokku) which will automatically read the `DATABASE_URL` variable on deploy and configure most of the app without human interaction required (thx to [automated_piwik_configs/12f_install.sh](https://github.com/TheNotary/piwik-dokku/blob/master/automated_piwik_configs/12f_install.sh)) If anyone can figure out how to make Heroku run scripts on deploy with access to ENV VARs, that would probably be the most ideal way to package this repo; so that configurations need to be made in only one place, at one point in time (e.g. app.json).

**Comment by dz0ny** [comment]
Heroku has support for release commands https://devcenter.heroku.com/articles/release-phase

**Comment by TheNotary** [comment]
I'm pretty burnt out from doing trail and air on the build stage at this point.  Feel free to borrow any of my code if you have freetime to hack on this and if you get something working on Heroku I'll look into converging repos.  

Also, I think you're clear to bump the piwik version up to `2.16.2`... I have a feeling a CVE will be coming out for `<= 2.16.0`.

**Comment by jasonfb** [comment]
After much trial & error I've concluded that Heroku is simply not right for Piwik. Piwik is very specialized and highly reliant on writing back to the file system. Trying to debug the Heroku docker build I also found very difficult, and was never able to correctly configure the installation in such a way that the settings in config.ini.php would actually persist (they would always be wiped away when the app restarted)

Instead, I attempted to spin up a simple EC2 instance on Amazon, installed PHP & nginx in basically no time flat — I had a few hiccups involving mbstring and some other PHP dependancies —  and then Piwik "just worked" out of the box the way I was expecting it to. Spinning it up on Amazon took me about 1 hour. Fighting with the Heroku settings and configuration took me several days and lead nowhere.

I am concluding that Heroku is the wrong platform for Piwik
